libweb3jsonrpc/UnixSocketServer.cpp

```diff
@@ -54,16 +57,45 @@ fs::path getIpcPathOrDataDir()
 		return getDataDir();
 	return path;
 }
+
+/**
+ * Waits for one of the file descriptors to become readable.
+ * @retval true One file descriptor became available.
+ * @retval false Most likely an error occured, or a timeout.
+ */
+static bool waitForReadable(std::initializer_list<int> sockets)
+{
+	fd_set in, out, err;
+	FD_ZERO(&in);
+	FD_ZERO(&out);
+	FD_ZERO(&err);
+
+	int const maxfd = *std::max_element(sockets.begin(), sockets.end());
+	for (int fd : sockets)
+		FD_SET(fd, &in);
+
+	return select(maxfd + 1, &in, &out, &err, nullptr) > 0;
+}
+
 }
 
 UnixDomainSocketServer::UnixDomainSocketServer(string const& _appId):
-	IpcServerBase((getIpcPathOrDataDir() / fs::path(_appId + ".ipc")).string().substr(0, c_socketPathMaxLength))
+	IpcServerBase((getIpcPathOrDataDir() / fs::path(_appId + ".ipc")).string().substr(0, c_socketPathMaxLength)),
+	m_exitChannel{-1, -1}
 {
+	if (pipe(m_exitChannel) < 0)
+		abort(); // I don't like. Replace me with throw.
```

Comment:
Throw exception here.
Declare it in `IpcServerBase.h` as
```
DEV_SIMPLE_EXCEPTION(FailedToCreatePipe);
```

Then throw here like
```
    BOOST_THROW_EXCEPTION(FailedToCreatePipe());
```